doc/stdenv/stdenv.chapter.md

```diff
@@ -991,13 +991,45 @@ Hook executed at the end of the fixup phase.
 
 If set to `true`, the standard environment will enable debug information in C/C++ builds. After installation, the debug information will be separated from the executables and stored in the output named `debug`. (This output is enabled automatically; you don’t need to set the `outputs` attribute explicitly.) To be precise, the debug information is stored in `debug/lib/debug/.build-id/XX/YYYY…`, where \<XXYYYY…\> is the \<build ID\> of the binary — a SHA-1 hash of the contents of the binary. Debuggers like GDB use the build ID to look up the separated debug information.
 
-For example, with GDB, you can add
+For example, to make GDB able to find debug information for the `socat` package and its dependencies, you can use the following
+`shell.nix`:
+
+```.nix
+let
+  pkgs = import ./. {
+    config = {};
+    overlays = [
+      (final: prev: {
+        ncurses = prev.ncurses.overrideAttrs { separateDebugInfo = true; };
+        readline = prev.readline.overrideAttrs { separateDebugInfo = true; };
+      })
+    ];
+  };
 
-```
-set debug-file-directory ~/.nix-profile/lib/debug
-```
+  myDebugInfoDirs = pkgs.symlinkJoin {
+    name = "myDebugInfoDirs";
+    paths = [
+      pkgs.glibc.debug
+      pkgs.ncurses.debug
+      pkgs.openssl.debug
+      pkgs.readline.debug
+    ];
```

Comment:
```suggestion
    paths = with pkgs; [
      glibc.debug
      ncurses.debug
      openssl.debug
      readline.debug
    ];
```

While minds are divided over `with`, I think it does help increasing clarity.

Reply:
Sure, I just got into the habit of not using with, but it is not an issue here.